htsinfer/infer_read_orientation.py

```diff
@@ -22,5 +21,45 @@ def infer(
         LIBTYPE string according to Salmon documentation, cf.
         https://salmon.readthedocs.io/en/latest/library_type.html
     """
+    #create .tmp directory in "inputDir" and retuns .tmp path
+    tmpDir = mkTmpDir(inputDir)
+    
+    # call functions of issues #25, 26, 27 and 34
+    
+    # delete .tmp
+    rmTmpDir(tmpDir)
+        
     # implement logic
     return "U"
+    
+
+def mkTmpDir(inputDir):
+    """Function to create a temporary directory for the storage of 
+    temporary data."""
+  
+    import os
+
+    # define the name of the directory to be created
+    tmpPath = inputDir + "/.tmp"
+  
+    try:
```

Comment:
Handling errors is better done as much upstream as possible in order to increase reusability of a function and separate concerns. So the `try - except` block should rather go in the calling context (or possibly even further up), not the function itself.

Another issue is that you are not re-raising the error, just printing an error message. This is a problem, because after printing, the program will just continue to execute _as if the directory were created_. But if we don't have a temporary directory, we can't continue with the code. So what you need to do is re-raise the error after logging your message. 
Also, never use `print()` for logging, use a proper (via the [`logging`[(https://docs.python.org/3/library/logging.html) module) instead. See `infer_single_paired.py` for an example. In this case, for the error log message, we can also raise the same (or a different/custom) error with a custom error message _from_ the raised error.

Finally, I highly recommend you to use ["f strings"](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python) for string interpolation, as they are shorter and much easier to read and write. In this case though, we don't know the name `mkdtemp()` generated, so there's nothing to interpolate. We'd just log that creation of the temporary directory failed.

All put together, your code would become (in `infer()`):

```py
import logging
from tempfile import mkdtemp
from typing import Union

LOGGER = logging.getLogger(__name__)


def infer(...):  # incomplete, of course use the whole function signature, docstring etc.

    # create temporary directory
    try:
        tmp_dir = mkdtemp(dir=__path__)
    except OSError as exc:
        raise OSError("Creation of temporary directory failed") from exc
    LOGGING.info(f"Created temporary directory {tmp_dir}.")
```